enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4919,6 +4929,416 @@ class AdjointGenerator
+              lookup(gutils->getNewFromOriginal(arg_incx), Builder2),    // incx
+              lookup(gutils->invertPointerM(arg_y, Builder2), Builder2), // [y]
+              lookup(gutils->getNewFromOriginal(arg_incy), Builder2)};   // incy
+          Builder2.CreateCall(
+              swapCall, args1,
+              gutils->getInvertedBundles(&call,
+                                         {ValueType::None, ValueType::Shadow,
+                                          ValueType::None, ValueType::Shadow,
+                                          ValueType::None},
+                                         Builder2, true));
+        }
+        if (has_diff_y) {
+          assert(has_diff_x && "Swapping with constant is not handled");
+        }
+      }
+    }
+
+    if ((funcName == "cblas_scopy" || funcName == "cblas_dcopy") &&
+        called->isDeclaration()) {
+      // void scopy(int n, float *x, int incx, float *y, int incy)
+      handled = true;
+      std::string axpyName = getBLASName(funcName, "axpy");
+      auto arg_n = call.getArgOperand(0), arg_x = call.getArgOperand(1),
+           arg_incx = call.getArgOperand(2), arg_y = call.getArgOperand(3),
+           arg_incy = call.getArgOperand(4);
+      auto type_n = arg_n->getType(), type_x = arg_x->getType(),
+           type_incx = arg_incx->getType(), type_y = arg_y->getType(),
+           type_incy = arg_incy->getType();
+      bool has_diff_x = !gutils->isConstantValue(arg_x),
+           has_diff_y = !gutils->isConstantValue(arg_y);
+      if (Mode == DerivativeMode::ReverseModeCombined ||
+          Mode == DerivativeMode::ReverseModeGradient) {
+        IRBuilder<> Builder2(call.getParent());
+        getReverseBuilder(Builder2);
+        auto new_n = lookup(gutils->getNewFromOriginal(arg_n), Builder2),
+             new_incx = lookup(gutils->getNewFromOriginal(arg_incx), Builder2),
+             new_incy = lookup(gutils->getNewFromOriginal(arg_incy), Builder2);
+        if (has_diff_x) {
+          if (has_diff_y) {
+            auto diff_x =
+                     lookup(gutils->invertPointerM(arg_x, Builder2), Builder2),
+                 diff_y =
+                     lookup(gutils->invertPointerM(arg_y, Builder2), Builder2);
+            auto axpyCall = module->getOrInsertFunction(
+                axpyName, voidType, type_n, scalarType, type_y, type_incy,
+                type_x, type_incx);
+            SmallVector<Value *, 6> args = {
+                new_n,  ConstantFP::get(scalarType, 1.0),
+                diff_y, new_incy,
+                diff_x, new_incx};
+            Builder2.CreateCall(
+                axpyCall, args,
+                gutils->getInvertedBundles(&call,
+                                           {ValueType::None, ValueType::Shadow,
+                                            ValueType::None, ValueType::Shadow,
+                                            ValueType::None},
+                                           Builder2, true));
```

Comment:
You should handle the case where x or y are inactive

Reply:
We don't need to handle either of these cases.
Case 1: x active, y inactive, then y have no gradient value to add to dx, so do nothing;
Case 2: x inactive, y active, then x have no gradient to be added to, so do nothing.
In summary: Unless x and y are both active, do nothing.